.github/workflows/automation-trigger-test.yml

```diff
@@ -1,18 +1,117 @@
+      - name: Upload build artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: release_framework
+          path: |
+            ./Branch_static.zip
+            
+  # This job
+  # 1.Downloads static framework build in job build-Brach-Static-framework and copies it to framework directory of qentelli-saas-sdk-testing-ios-testbed repo
+  # 2.Builds testbed app.
+  # 3.Installs Certificates and Provisioning profile for signing App and creating IPA. Final IPA is uploaded as an artifact.
+  build-Automation-testbed-App:
+    runs-on: macos-latest
+    needs: [build-Branch-Static-framework]
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+        with:
+          repository: BranchMetrics/qentelli-saas-sdk-testing-ios-testbed
+          ref: test-dynamic-linking
+          token: ${{ secrets.BRANCHLET_ACCESS_TOKEN_PUBLIC }}
+      - name: Install the Apple certificate and provisioning profile
+        env:
+          BUILD_CERTIFICATE_BASE64: ${{ secrets.BUILD_CERTIFICATE_BASE64 }}
+          P12_PASSWORD: ${{ secrets.P12_PASSWORD }}
+          BUILD_PROVISION_PROFILE_BASE64: ${{ secrets.BUILD_PROVISION_PROFILE_BASE64 }}
+          KEYCHAIN_PASSWORD: ${{ secrets.KEYCHAIN_PASSWORD }}
+        run: |
+          # create variables
+          CERTIFICATE_PATH=$RUNNER_TEMP/build_certificate.p12
+          KEYCHAIN_PATH=$RUNNER_TEMP/app-signing.keychain-db
+          PP_PATH=$RUNNER_TEMP/build_pp.mobileprovision
+
+          # import certificate 
+          echo -n "$BUILD_CERTIFICATE_BASE64" | base64 --decode -o $CERTIFICATE_PATH
+          echo -n "$BUILD_PROVISION_PROFILE_BASE64" | base64 --decode -o $PP_PATH
+
+          # create temporary keychain
+          security create-keychain -p "$KEYCHAIN_PASSWORD" $KEYCHAIN_PATH
+          security set-keychain-settings -lut 21600 $KEYCHAIN_PATH
+          security unlock-keychain -p "$KEYCHAIN_PASSWORD" $KEYCHAIN_PATH
+
+          # import certificate to keychain
+          security import $CERTIFICATE_PATH -P "$P12_PASSWORD" -A -t cert -f pkcs12 -k $KEYCHAIN_PATH
+          security list-keychain -d user -s $KEYCHAIN_PATH
+
+          # apply provisioning profile
+          mkdir -p ~/Library/MobileDevice/Provisioning\ Profiles
+          cp $PP_PATH ~/Library/MobileDevice/Provisioning\ Profiles
+          
+      - name: Download Branch SDK lib, build testbed app and create IPA File
+        uses: actions/download-artifact@v3
+      - name : Unzip and copy folder
+        run: |
+            unzip release_framework/Branch_static.zip -d ./Framework/
+            xcodebuild -workspace DeepLinkDemo.xcworkspace -scheme DeepLinkDemo  -sdk iphoneos archive -archivePath ./IPA/DeepLinkDemo.xcarchive
+            xcodebuild -exportArchive -archivePath ./IPA/DeepLinkDemo.xcarchive -exportOptionsPlist IPA/Info.plist -exportPath IPA/
+      - name: Upload build artifacts
+        uses: actions/upload-artifact@v3
```

Comment:
Instead of uploading ipa as an artifact, we can do one of these : 
- use cache which is in general faster than uploading & downloading artifacts to share data between jobs
- upload the ipa to the browserstack cloud with a custom id or upload and save the returned `app_url`. This way we can break the browserstack test execution into independently executable / failable steps. We can fail fast in case of an issue.
https://www.browserstack.com/docs/app-automate/appium/upload-app-from-filesystem
https://www.browserstack.com/app-live/rest-api